packages/typespec-ts/src/modular/helpers/operationHelpers.ts

```diff
@@ -435,8 +435,12 @@ function extractPagingType(type: Type, itemName?: string): Type | undefined {
   if (!itemName) {
     return undefined;
   }
-  const prop = (type.properties ?? [])
-    ?.filter((prop) => prop.restApiName === itemName)
+  const allProperties = [
+    ...(type.properties ?? []),
```

Comment:
Could you add a UT in https://github.com/Azure/autorest.typescript/blob/main/packages/typespec-ts/test/modularUnit/operations.spec.ts#L606?

Reply:
https://github.com/Azure/autorest.typescript/pull/2764/commits/d4e21d788abff3eba0768e9d93fe0eb26885d093